### ISSUE TITLE
`AliasResolutionError` with a suspected fix.

### ISSUE THREAD

**Issue by deepandas11** [mkdocstrings/griffe#114]
**Describe the bug**
I encountered this when using `mkdocstrings` on a fairly large mono-repo style codebase with `mkdocs-monorepo`. 

The python handler from `mkdocstrings` would eventually trickle down to `griffe.loader.resolve_module_aliases` and would fail for a certain file. 
<details>
<summary>Traceback</summary>

```bash
    data: CollectorItem = handler.collect(identifier, options)
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/mkdocstrings_handlers/python/handler.py", line 199, in collect
    unresolved, iterations = loader.resolve_aliases(implicit=False, external=False)
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/griffe/loader.py", line 217, in resolve_aliases
    next_resolved, next_unresolved = self.resolve_module_aliases(module, implicit, external)
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/griffe/loader.py", line 357, in resolve_module_aliases
    sub_resolved, sub_unresolved = self.resolve_module_aliases(
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/griffe/loader.py", line 357, in resolve_module_aliases
    sub_resolved, sub_unresolved = self.resolve_module_aliases(
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/griffe/loader.py", line 357, in resolve_module_aliases
    sub_resolved, sub_unresolved = self.resolve_module_aliases(
  [Previous line repeated 1 more time]
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/griffe/loader.py", line 335, in resolve_module_aliases
    if not implicit and not member.is_explicitely_exported:
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/griffe/dataclasses.py", line 827, in __getattr__
    attr = getattr(self.target, name)
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/griffe/dataclasses.py", line 936, in target
    self.resolve_target()
  File "/Users/deepandas/.pyenv/versions/3.8.13/envs/mkdocs_test/lib/python3.8/site-packages/griffe/dataclasses.py", line 964, in resolve_target
    raise AliasResolutionError(self.target_path) from error
```
</details>

However, I followed a hunch and was able to fix it magically by monkeypatching a fix in `expressions`:

https://github.com/mkdocstrings/griffe/blob/2c3b0cea0c9a216fc0a25445e0dbe9619a15a199/src/griffe/expressions.py#L32-L33

I simply added a path to allow `str` types 
```python
def __eq__(self, other: Name | Expression | str) -> bool:  # type: ignore[override]
        if isinstance(other, str):
            return self.full == other
        return self.full == other.full  # noqa: WPS437
```

**To Reproduce**
I haven't been able to find a demonstrable repro on a smaller scale, which makes me believe this could also stem from a weird import order issue from the large codebase I am trying to handle. 

**Expected behavior**

**Screenshots**

**System (please complete the following information):**
- `griffe` version: 0.24.1
- `mkdocstrings` version: 0.8.0
- Python version: 3.8.13
- OS: MacOS

**Additional context**
I haven't played around with the library too much and wasn't able to figure out the underlying reason. Looking for more context into this and if possible, to accommodate the simple fix?

**Comment by pawamoy** [comment]
Hello, thanks a lot for the detailed report, for the time you took to investigate, and for the suggested fix :rocket:!

I'm very curious as to how you found that this change would fix the issue. I guess you ran a debugging session?
I also guess your monorepo is not public, and therefore I cannot debug on my end as well, or try to find a MRE :sweat_smile:?

**Comment by deepandas11** [comment]
Thanks for the swift response! Appreciate you looking into it.

> I'm very curious as to how you found that this change would fix the issue. I guess you ran a debugging session?

Honestly, I didn't spend a lot of time into it, and my hunch originated from the `member.is_explicitly_exported` check on the traceback. Following that trail led me to this line in `dataclasses.py`:

https://github.com/mkdocstrings/griffe/blob/2c3b0cea0c9a216fc0a25445e0dbe9619a15a199/src/griffe/dataclasses.py#L409

The type hints in the initializer suggests `exports` is `set[str]` or `list[str | Name]`, qualifying the `__eq__(self, other)` on `Name` for the `list[Name]` case when `other` is of type `Name`, but not for `str`. 





> I also guess your monorepo is not public, and therefore I cannot debug on my end as well, or try to find a MRE 😅?

Unfortunately, yes, it's a private repo, and I did put in some effort into setting up a minimal repro, but that didn't work. I am hoping the traceback, and my line of thought above could assist you in some way?

**Comment by pawamoy** [comment]
Should be fixed by 37ae0a2, thanks again @deepandas11. I'll try to pack a few more things before the next release.
